docs/src/web-client/foreign_procedure_invocation_tutorial.md

```diff
@@ -0,0 +1,614 @@
+
+`package.json`
+
+```json
+  "scripts": {
+    "dev": "next dev",
+    ...
+  }
+```
+
+## Step 2: Edit the `app/page.tsx` file
+
+Add the following code to the `app/page.tsx` file. This code defines the main page of our web application:
+
+```tsx
+"use client";
+import { useState } from "react";
+import { foreignProcedureInvocation } from "../lib/foreignProcedureInvocation";
+
+export default function Home() {
+  const [isFPIRunning, setIsFPIRunning] = useState(false);
+
+  const handleForeignProcedureInvocation = async () => {
+    setIsFPIRunning(true);
+    await foreignProcedureInvocation();
+    setIsFPIRunning(false);
+  };
+
+  return (
+    <main className="min-h-screen flex items-center justify-center bg-gradient-to-br from-gray-900 via-gray-800 to-black text-slate-800 dark:text-slate-100">
+      <div className="text-center">
+        <h1 className="text-4xl font-semibold mb-4">Miden FPI Web App</h1>
+        <p className="mb-6">Open your browser console to see WebClient logs.</p>
+
+        <div className="max-w-sm w-full bg-gray-800/20 border border-gray-600 rounded-2xl p-6 mx-auto flex flex-col gap-4">
+          <button
+            onClick={handleForeignProcedureInvocation}
+            className="w-full px-6 py-3 text-lg cursor-pointer bg-transparent border-2 border-orange-600 text-white rounded-lg transition-all hover:bg-orange-600 hover:text-white"
+          >
+            {isFPIRunning
+              ? "Working..."
+              : "Foreign Procedure Invocation Tutorial"}
+          </button>
+        </div>
+      </div>
+    </main>
+  );
+}
+```
+
+## Step 3: Create the Foreign Procedure Invocation Implementation
+
+Create the file `lib/foreignProcedureInvocation.ts` and add the following code.
+
+```bash
+mkdir -p lib
+touch lib/foreignProcedureInvocation.ts
+```
+
+Copy and paste the following code into the `lib/foreignProcedureInvocation.ts` file:
+
+```ts
+// lib/foreignProcedureInvocation.ts
+export async function foreignProcedureInvocation(): Promise<void> {
+  if (typeof window === "undefined") {
+    console.warn("foreignProcedureInvocation() can only run in the browser");
+    return;
+  }
+
+  // dynamic import → only in the browser, so WASM is loaded client‑side
+  const {
+    AccountBuilder,
+    AccountComponent,
+    AccountId,
+    AccountType,
+    AssemblerUtils,
+    StorageSlot,
+    TransactionKernel,
+    TransactionRequestBuilder,
+    TransactionScript,
+    TransactionScriptInputPairArray,
+    ForeignAccount,
+    AccountStorageRequirements,
+    WebClient,
+    AccountStorageMode,
+  } = await import("@demox-labs/miden-sdk");
+
+  const nodeEndpoint = "https://rpc.testnet.miden.io:443";
+  const client = await WebClient.createClient(nodeEndpoint);
+  console.log("Current block number: ", (await client.syncState()).blockNum());
+
+  // -------------------------------------------------------------------------
```

Comment:
One question arises: Does it make sense to deploy a counter contract already on testnet from which this tutorial just reads the state? 

If we first deploy a counter contract, it only complicates things for the developer, since 

- we only want to explain FPI in this tutorials 
- we should not use the WebClient to create an deploy "complex" smart contracts but rather to interact with it

So ideally, we should create a note (if its simpler than an account) that reads the already deployed contract state. We can think of a note that transfers asset A if the counter contract state is an odd number and sends asset B back and vice versa. 

Does that make sense?